### DISCUSSION TITLE
How to manually create a selection

### DISCUSSION THREAD

**Discussion by ceberterreche** [facebook/lexical#5475] [Q&A]
Good morning,

I would like to know if it is possible to create a selection like this: createRangeSelection(startIndex, endIndex).

My goal is to convert text format similar to markdown with tags like this: "this is my [BOLD]brilliant[/BOLD] text" to lexical format.

In this example, I would like to create a paragraph and apply bold to the part that has the [BOLD] tag.

Thx

**Comment by moy2010** [comment]
The `setTextNoderange` method in `RangeSelection` seems to be what you are looking for:

```
setTextNodeRange(anchorNode: TextNode, anchorOffset: number, focusNode: TextNode, focusOffset: number): void;
```

Note that this requires that you already have a valid instance of the Lexical editor.
